src/auth/auth.controller.ts

```diff
@@ -2,31 +2,39 @@ import { Body, Controller, Post, UseGuards } from '@nestjs/common';
 import { AuthService } from './auth.service';
 
 import { UserService } from 'src/users/users.service';
-
-import { LoginDTO } from './auth-login.dto';
-import { AuthGuard } from '@nestjs/passport/dist/auth.guard';
-import { UserRegisterDTO } from 'src/dtos/users.dto';
+import { ApiResponse, ApiTags } from '@nestjs/swagger';
+import { LoginDto, RegistrationDTO } from 'src/users/users.dto';
 
 @Controller('auth')
+@ApiTags('Auth')
 export class AuthController {
   constructor(
     private userService: UserService,
     private authService: AuthService,
   ) {}
 
-  @Post('register')
-  async register(@Body() registerDTO: UserRegisterDTO) {
-    const user: any = await this.userService.create(registerDTO);
-    const payload = user.email;
+  @ApiResponse({
+    status: 200,
+    description: 'A user has been succesfully registered',
+  })
+  @UseGuards()
+  @Post('/register')
+  async register(@Body() body: RegistrationDTO) {
+    const token = await this.authService.signPayload({ email: body.email });
+
+    const user: any = await this.userService.createUser(body);
 
-    const token = await this.authService.signPayload(payload);
     return { user, token };
   }
 
+  @ApiResponse({
+    status: 200,
+    description: 'User Logged in successfully',
+  })
   @Post('/login')
-  @UseGuards(AuthGuard('jwt'))
-  async login(@Body() loginDTO: LoginDTO) {
-    const user: any = await this.userService.findByLogin(loginDTO);
+  //@UseGuards(AuthGuard('jwt'))
+  async login(@Body() body: LoginDto) {
+    const user: any = await this.userService.findByLogin(body);
 
```

Comment:
Where in the world do people log in with only email and without a password? This makes your system vulnerable

Reply:
its both that is there boss